### DISCUSSION TITLE
Mogrify conversion of SVG with text on path fails

### DISCUSSION THREAD

**Discussion by Dlap2** [ImageMagick/ImageMagick#2915] [Help]
I have an application that converts files from SVG to JPG or PNG.  I build the SVG images with Inkscape and when I mogrify them, it works properly unless I set the text along a path - in which case the path shows but the text is not there.  Any suggestions?

I have tried saving in Inkscape SVG, Plain SVG, and Optimized SVG.  When I try to save as HTML, I get a Python error.

**Comment by fmw42** [comment]
Does it do the same using convert (IM 6) or magick (IM 7)?  Which renderer is being use by ImageMagick (Inkscape, RSVG, MSVG)?  Put -verbose in your convert or magick command to see which is being used.

**Reply by Dlap2** [reply]
I will check Convert, but -verbose doesn't seem to be showing me the renderer...


C:\Users\David>"d:\program files\imagemagick-7.0.10-q16-hdri\magick" mogrify -verbose -path c:\tempgeorgia\jpg -format jpg C:\tempgeorgia\*.svg
C:\tempgeorgia\template.svg SVG 231x399 231x399+0+0 16-bit sRGB 778777B 0.172u 0:00.180
C:\tempgeorgia\template.svg=>c:\tempgeorgia\jpg\template.jpg SVG 231x399 231x399+0+0 16-bit sRGB 41954B 0.016u 0:00.027

**Comment by Dlap2** [comment]
Convert does the same thing - tried converting to JPG and PNG.  Path shows, but not text.  Text not on a path shows.

**Comment by Dlap2** [comment]
Inkscape command line works with text on path.

**Comment by snibgo** [comment]
A long time ago, when dealing with [SVG text](http://im.snibgo.com/svgtext.htm) on paths, I wrote:

> However, the formatted text facilities used here don't work when Inkscape is called from IM, so for these I need to call Inkscape directly.

I don't know if IM still has that problem. From your question, I suppose it does.

**Comment by Dlap2** [comment]
Is it possible to force the use of a different renderer?

**Reply by fmw42** [reply]
Yes, but Inkscape will most likely be the best and should work if installed on current IM versions. 

If you want to use MSVG then 

`magick MSVG:image.svg image.png`

If you need RSVG, then you would have to install the RSVG delegate and modify your delegates.xml file.

Try

`magick -verbose image.svg image.png`

You should get a long text output that ultimately will tell what renderer is being used.

**Comment by Dlap2** [comment]
Thanks for your help.  MSVG: still doesn't have the text, but it took a lot longer than without MSVG: .  Don't know if 

C:\tempGeorgia>"d:\program files\imagemagick-7.0.10-q16-hdri\magick" -verbose template.svg template.png
template.svg SVG 231x399 231x399+0+0 16-bit sRGB 778777B 0.188u 0:00.191
template.svg=>template.png SVG 231x399 231x399+0+0 16-bit sRGB 778777B 0.250u 0:00.113

tells you the renderer used.  The advantage of Imagemagick is tht it is much faster than calling InkScape for each file (maybe 20 times faster). 

Any other suggestions?

**Reply by fmw42** [reply]
What is your OS and exact version of IM (7.0.10.x) from `magick -version`

**Comment by Dlap2** [comment]
Windows 10 and 
Version: ImageMagick 7.0.10-45 Q16 x64 2020-11-30

**Comment by Dlap2** [comment]
Updated to 1.01 and it didn't change anything.

**Comment by snibgo** [comment]
If "-verbose" doesn't say IM is using Inkscape, then IM isn't using Inkscape.

If you put Inkscape on your system path, then IM should use Inkscape.

If IM uses Inkscape but the text still isn't on the path, then try calling Inkscape directly from the command line. Does that put the text on the path?

**Reply by fmw42** [reply]
If you have Inkscape if it is not on your system PATH, then ImageMagick may not be able to find it to use it for rendering your SVG. Make sure the path to Inkscape is in your PATH environment variable or however that works on Windows. Then try putting -verbose again in your magick command (not mogrify).

**Comment by Dlap2** [comment]
Inkscape does put the text on the path.

**Comment by snibgo** [comment]
Okay, then I suggest that is your solution.
